dna/models/base_models.py

```diff
@@ -65,12 +76,18 @@ def fit(
         if output_dir is not None:
             model_save_path = os.path.join(output_dir, 'model.pt')
 
+        train_data, validation_data = self._get_validation_split(train_data, validation_ratio, self._validation_split_seed)
+
         train_data_loader = self._get_data_loader(train_data, batch_size, drop_last, shuffle=True)
         validation_data_loader = None
         min_loss_score = np.inf
+        min_loss_epoch = -1
         if validation_data is not None:
             validation_data_loader = self._get_data_loader(validation_data, batch_size, drop_last=False, shuffle=False)
 
+        if patience < 1:
+            patience = np.inf
+
         for e in range(n_epochs):
```

Comment:
It took me a second to realize what e was (error / epochs) but maybe it's just been a long day haha.  Not that you did that change, just saying.

Reply:
Haha, perhaps that's not the best variable name since `e` is often used for errors.